contexts/compound-and-earn-context.js

```diff
@@ -377,7 +401,72 @@ export function CompoundAndEarnProvider({ children }) {
   };
 
-  const deposit = async (item, amount, onlyGauge = false) => {
+  const zapIntoSnowglobe = async (amount, baseTokenAddress, item, isNativeAVAX) => {
+    const zapperAddress = getZapperContract(item);
+    const zappersContract = new ethers.Contract(zapperAddress, SNOWGLOBE_ZAPPERS_ABI, library.getSigner());
+    if(!isNativeAVAX){ 
+      const estimateSwap = await zappersContract.estimateSwap(item.address, baseTokenAddress, amount);
+
+      //1% slippage
+      const amountMinToken = estimateSwap.swapAmountOut.sub(estimateSwap.swapAmountOut.div(100));
+
+      const zapTx = await zappersContract.zapIn(item.address, amountMinToken, baseTokenAddress, amount);
+      return zapTx;
+    } else { 
+      //if the pair uses WAVAX we want to go through it because it costs less gas!!
+      let hasWAVAX, WAVAXPos;
+      if(item.token0.address === WAVAX) {
+        hasWAVAX = true;
+        WAVAXPos = 0;
+      } else if (item.token1.address === WAVAX) {
+        hasWAVAX = true;
+        WAVAXPos = 1;
+      }
+
+      if(hasWAVAX){
+        const estimateSwap = await zappersContract.estimateSwap(item.address, item[`token${WAVAXPos}`].address, amount);
+
+        //1% slippage
+        const amountMinToken = estimateSwap.swapAmountOut.sub(estimateSwap.swapAmountOut.div(100));
+
+        const zapTx = await zappersContract.zapInAVAX(item.address, amountMinToken, WAVAX, {value: amount});
+        return zapTx;
+      } else {
+        let routerAddress;
+        switch(item.source) {
+          case "Trader Joe": case "Axial":
+            routerAddress = CONTRACTS.ROUTER_TRADERJOE;
+            break;
+          case "Pangolin":
+            routerAddress = CONTRACTS.ROUTER_PANGOLIN;
+            break;
+          default:
+            throw new Error("Router not found for this pool");
+        }
+        //simulate a swap between WAVAX and TOKEN0 to know how much it is worth
+        const routerContract = new ethers.Contract(routerAddress,AMM_ROUTER_ABI,library.getSigner());
+
+        let amountOut, tokenPos
+        try {
+          [, amountOut] = await routerContract.getAmountsOut(amount,[WAVAX, item.token0.address]);
+          tokenPos = 0;
+        } catch (error) {
+          //if there`s no path WAVAX/TOKEN0 we try token1
+          [, amountOut] = await routerContract.getAmountsOut(amount,[WAVAX, item.token1.address]);
+          tokenPos = 1;
+        }
+
+        const estimateSwap = await zappersContract.estimateSwap(item.address, item[`token${tokenPos}`].address, amountOut);
+        //1% slippage
+        const amountMinToken = estimateSwap.swapAmountOut.sub(estimateSwap.swapAmountOut.div(100));
+        const zapTx = await zappersContract.zapInAVAX(item.address, amountMinToken, item[`token${tokenPos}`].address, {value: amount});
+        
+        return zapTx;
+      }
+    }
+  }
+
+  const deposit = async (item, amount, addressFromZapper, onlyGauge = false, isNativeAVAX = false) => {
```

Comment:
We need to check if this function is not used elsewhere, because it's declaration changed